pkgs/development/compilers/swift/swiftpm/default.nix

```diff
@@ -0,0 +1,469 @@
+{ lib
+, callPackage
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+, git
+, swift
+, Foundation
+, XCTest
+, sqlite
+, ncurses
+, substituteAll
+, runCommandLocal
+, makeWrapper
+, DarwinTools # sw_vers
+, cctools # vtool
+, xcbuild
+, CryptoKit
+, LocalAuthentication
+}:
+
+let
+
+  inherit (swift) swiftOs swiftModuleSubdir swiftStaticModuleSubdir;
+  sharedLibraryExt = stdenv.hostPlatform.extensions.sharedLibrary;
+
+  # Common attributes for the bootstrap swiftpm and the final swiftpm.
+  commonAttrs = rec {
+    # Releases are made as part of the Swift toolchain, so versions should match.
+    version = "5.7";
+    src = fetchFromGitHub {
+      owner = "apple";
+      repo = "swift-package-manager";
+      rev = "swift-${version}-RELEASE";
+      hash = "sha256-MZah+/XfeK46YamxwuE3Kiv+u5bj7VmjEh6ztDF+0j4=";
+    };
+    nativeBuildInputs = [ makeWrapper ];
+    # Required at run-time for the host platform to build package manifests.
+    propagatedBuildInputs = [ Foundation ];
```

Comment:
Maybe this needs a better comment. It's swiftpm that requires it at run-time. So, a derivation that uses swiftpm to build something needs Foundation as a `nativeBuildInput`, and this makes that automatic when swiftpm is already a `nativeBuildInput`. It's needed for when swiftpm compiles `Package.swift`.

Reply:
Why would a run-time dependency of a tool need to be a build input to a dependent? Shouldn't run-time dependencies always be part of a tool's closure anyway?

Reply:
Foundation needs to be a nativeBuildInput of a package using swiftpm, because swiftpm performs a build step for `Package.swift` targeting the build platform. Having it the closure is not enough because only buildInputs are processed by setup hooks, which add the necessary search paths to compiler flags.

Reply:
OK, definitely confusing but I think I follow your reasoning.